pkg/kv/kvserver/replica_send.go

```diff
@@ -277,6 +277,23 @@ func (r *Replica) maybeAddRangeInfoToResponse(
 	}
 }
 
+func (r *Replica) maybeThrottleRequestPreLatch(ctx context.Context, ba *roachpb.BatchRequest) bool {
+	switch {
+	case ba.IsSingleScanInterleavedIntentsRequest():
+		batcheval.ThrottleScanInterleavedIntents(ctx, r.StoreID())
+		return true
+	default:
+		return false
+	}
+}
+
+func (r *Replica) cleanupRequestThrottling(_ context.Context, ba *roachpb.BatchRequest) {
+	if !ba.IsSingleScanInterleavedIntentsRequest() {
+		panic("no throttling cleanup for a non-ScanInterleavedIntents request")
```

Comment:
2¢: I donno why I'm slightly uneasy about the panic, as well as the map lookup, plus being getOrCreate here (we would never want to create, then release). I might vote to have maybeThrottle just return a limiter.Reservation and if the caller gets a non-nil reservation, they have to defer releasing it.